apps/crawler/src/processors/extrinsic-processor.ts

```diff
@@ -0,0 +1,170 @@
+import { Connection, Repository } from 'typeorm';
+import { InjectRepository } from '@nestjs/typeorm';
+import { ExtrinsicHandlerContext } from '@subsquid/substrate-processor';
+import { Extrinsic } from '@entities/Extrinsic';
+import { ExtrinsicNames } from '@common/constants';
+import { ScanProcessor } from './scan-processor';
+import { SdkService } from '../sdk.service';
+import { UtilsService } from '@common/utils/utils.service';
+import { ProcessorConfigService } from '../processor.config.service';
+import { Event } from '@entities/Event';
+
+interface IExtrinsicOwner {
+  value: { id?: string; substrate?: string; ethereum?: string };
+}
+
+const transferMethods = [
+  'transfer',
+  'transferAll',
+  'transferKeepAlive',
+  'vestedTransfer',
+];
+
+@Injectable()
+export class ExtrinsicProcessor {
+  name = 'extrinsic';
+  private logger: Logger;
+  private processor: ScanProcessor;
+
+  constructor(
+    @InjectRepository(Extrinsic)
+    private modelRepository: Repository<Extrinsic>,
+    private utils: UtilsService,
+    protected connection: Connection,
+    protected sdkService: SdkService,
+    private processorConfigService: ProcessorConfigService,
+  ) {
+    this.processor = new ScanProcessor(
+      this.name,
+      this.connection,
+      processorConfigService.getDataSource(),
+      processorConfigService.getRange(),
+      processorConfigService.getTypesBundle(),
+    );
+
+    this.logger = new Logger('ExtrinsicProcessor');
+
+    Object.values(ExtrinsicNames).forEach((name) => {
+      this.processor.addExtrinsicHandler(name, async (ctx) => {
+        await this.upsertHandler(ctx, name);
+      });
+    });
+
+    this.logger.log('Starting processor...');
+  }
+
+  private async upsertHandler(
+    ctx: ExtrinsicHandlerContext,
+    eventName: string,
+  ): Promise<void> {
+    const { height, timestamp } = ctx.block;
+
+    const log = {
+      eventName,
+      blockNumber: height,
+      blockTimestamp: timestamp,
+    };
+
+    try {
+      const data = await this.getData(ctx);
+
+      if (data) {
+        await this.modelRepository.upsert(data, [
+          'block_number',
+          'extrinsic_index',
+        ]);
+      }
+
+      this.logger.verbose({ ...log });
+    } catch (err) {
+      this.logger.error({ ...log, error: err.message });
+    }
+  }
+
+  private async getData(ctx: ExtrinsicHandlerContext) {
+    const { block, extrinsic } = ctx;
+    const events = await this.getExtrinsicEvents(ctx);
+
+    let to_owner: null | string = null;
+    if (transferMethods.includes(extrinsic.method)) {
+      const args = extrinsic.args;
+      const ownerData = args[0] as IExtrinsicOwner;
+      if (ownerData) {
+        to_owner =
+          ownerData.value.id ||
+          ownerData.value.substrate ||
+          ownerData.value.ethereum ||
+          null;
+      }
+    }
+
+    return {
+      block_number: block.height,
+      extrinsic_index: extrinsic.indexInBlock,
+      is_signed: true,
```

Comment:
I'm not shure it's correct. Some extrinsics are not signed. For example timestamp.set. Why don't check extrinsic.isSigned field?